src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -1153,6 +1156,12 @@ namespace QgsWms
           break;
         }
       }
+
+      if ( !validLayer )
+      {
+        throw QgsBadRequestException( QStringLiteral( "LayerNotDefined" ),
+                                      QStringLiteral( "Layer '%1' not found" ).arg( queryLayer ) );
```

Comment:
I'm always unsure, should such strings be translated or not?

Reply:
None of the `QgsBadRequestException` raised by the server are currently translated. @rldhont is there a reason?

I know that OGC tests check the error code (like `LayerNotDefined`) returned by the server to validate the compliancy but beyond that, I don't know...

Reply:
The first one `LayerNotDefined` has not to be translated because it's a code to identify exception.